tests/php/View/Parsers/ShortcodeParserTest.php

```diff
@@ -79,37 +79,54 @@ public function testNotRegisteredShortcode()
         );
     }
 
-    public function testSimpleTag()
+    public function simpleTagDataProvider()
     {
-        $tests = [
-            '[test_shortcode]',
-            '[test_shortcode ]', '[test_shortcode,]', '[test_shortcode, ]' . '[test_shortcode/]', '[test_shortcode /]', '[test_shortcode,/]', '[test_shortcode, /]'
+        return [
+            ['[test_shortcode]'],
+            ['[test_shortcode ]'],
+            ['[test_shortcode,]'],
+            ['[test_shortcode, ][test_shortcode/]'],
+            ['[test_shortcode /]'],
+            ['[test_shortcode,/]'],
+            ['[test_shortcode, /]']
         ];
+    }
 
-        foreach ($tests as $test) {
-            $this->parser->parse($test);
-
-            $this->assertEquals([], $this->arguments, $test);
-            $this->assertEquals('', $this->contents, $test);
-            $this->assertEquals('test_shortcode', $this->tagName, $test);
-        }
+    /**
+     * @dataProvider simpleTagDataProvider
+     */
+    public function testSimpleTag($test)
+    {
+        $this->parser->parse($test);
+        $this->assertEquals([], $this->arguments, $test);
+        $this->assertEquals('', $this->contents, $test);
+        $this->assertEquals('test_shortcode', $this->tagName, $test);
     }
 
-    public function testOneArgument()
+    public function oneArgumentDataProvider()
     {
-        $tests =  [
-            '[test_shortcode foo="bar"]', '[test_shortcode,foo="bar"]',
-            "[test_shortcode foo='bar']", "[test_shortcode,foo='bar']",
-            '[test_shortcode  foo  =  "bar"  /]', '[test_shortcode,  foo  =  "bar"  /]'
+        return [
+            ['[test_shortcode foo="bar"]'],
+            ['[test_shortcode,foo="bar"]'],
+            ["[test_shortcode foo='bar']"],
+            ["[test_shortcode,foo='bar']"],
+            ["[test_shortcode foo=bar]"],
+            ["[test_shortcode,foo=bar]"],
```

Comment:
Refactored `testOneArgument` to use a data provider. The two no quotes args are new tests